log/value.go

```diff
@@ -0,0 +1,68 @@
+package log
+
+import (
+	"time"
+
+	"gopkg.in/stack.v1"
+)
+
+// BindValues returns a slice with all value elements (odd indexes) that
+// implement Valuer replaced with the result of calling their Value method. If
+// no value elements implement Valuer, the original slice is returned.
+func BindValues(keyvals ...interface{}) []interface{} {
+	if !containsValuer(keyvals) {
+		return keyvals
+	}
+
+	bound := make([]interface{}, len(keyvals))
+	copy(bound, keyvals)
+	for i := 1; i < len(bound); i += 2 {
+		if v, ok := bound[i].(Valuer); ok {
+			bound[i] = v.Value()
+		}
+	}
+
+	return bound
+}
```

Comment:
That will also inspect keys for Valuers, which wasn't the intention (but open for debate if you think we should do that).

Also, I tried something like that during development. If memory serves is was a little slower, perhaps due to the extra bounds checking in the else clause.